onnxruntime/contrib_ops/cuda/bert/attention_impl.cu

```diff
@@ -126,30 +257,32 @@ Status QkvToContext(
     if (data.bias == nullptr) {
       // gemm_buffer should be BxSx3xNxH => qkv: 3xBxNxSxH
       ORT_ENFORCE(qk_head_size == v_head_size);
+      unsigned qkv_mask = (past_present_share_buffer ? 1 : 7);
       ORT_RETURN_IF_ERROR(LaunchTransQkv(stream, 3, sequence_length, batch_size, qk_head_size, num_heads,
```

Comment:
>LaunchTransQkv [](http://example.com/codeflow?start=26&length=14)

Do you need qkv_mask? How about to call LaunchTransQkv with num_of_matrix equal to 1 if past_present_share_buffer.

Reply:
do not want to change too much on existing implementation. 3 is related with next stride to compute next position. There exists other ways with add more changes to this function, may worth to try if found useful.